src/main/scala/chisel3/ltl/LTL.scala

```diff
@@ -0,0 +1,391 @@
+  */
+object Delay {
+  def apply(): SequenceAtom = DelayAtom(1, Some(1))
+  def apply(delay: Int): SequenceAtom = DelayAtom(delay, Some(delay))
+  def apply(min:   Int, max: Int): SequenceAtom = DelayAtom(min, Some(max))
+  def apply(min:   Int, max: Option[Int]): SequenceAtom = DelayAtom(min, max)
+}
+
+/** A Linear Temporal Logic (LTL) sequence. */
+sealed trait Sequence extends Property {
+
+  /** See `Sequence.delay`. */
+  def delay(delay: Int = 1): Sequence = Sequence.delay(this, delay)
+
+  /** See `Sequence.delayRange`. */
+  def delayRange(min: Int, max: Int): Sequence = Sequence.delayRange(this, min, max)
+
+  /** See `Sequence.delayAtLeast`. */
+  def delayAtLeast(delay: Int): Sequence = Sequence.delayAtLeast(this, delay)
+
+  /** See `Sequence.concat`. */
+  def concat(other: Sequence): Sequence = Sequence.concat(this, other)
+
+  /** See `Sequence.and`. */
+  def and(other: Sequence): Sequence = Sequence.and(this, other)
+
+  /** See `Sequence.or`. */
+  def or(other: Sequence): Sequence = Sequence.or(this, other)
+
+  /** See `Sequence.clock`. */
+  override def clock(clock: Clock): Sequence = Sequence.clock(this, clock)
+
+  /** See `Property.implication`. */
+  def implication(prop: Property): Property = Property.implication(this, prop)
+
+  /** See `Property.implication`. */
+  def implicationNonOverlapping(prop: Property): Property = Property.implicationNonOverlapping(this, prop)
+
+  // Convenience operators.
+
+  /** Equivalent to `|->` in SVA. */
+  def |->(prop: Property): Property = this.implication(prop)
+
+  /** Equivalent to `|=>` in SVA. */
+  def |=>(prop: Property): Property = this.implicationNonOverlapping(prop)
+
+  /** Equivalent to `a ## b` in SVA. */
+  def ###(other: Sequence): Sequence = this.concat(other.delay())
+
+  /** Equivalent to `a ##[*] b` in SVA. */
+  def ##*(other: Sequence): Sequence = this.concat(other.delayAtLeast(0))
+
+  /** Equivalent to `a ##[+] b` in SVA. */
+  def ##+(other: Sequence): Sequence = this.concat(other.delayAtLeast(1))
+}
+
+/** Prefix-style utilities to work with sequences.
+  *
+  * This object exposes the primary API to create and compose sequences from
+  * booleans and shorter sequences.
+  */
+object Sequence {
+
+  /** Implicitly wraps a `Bool` and allows it to be used as a sequence or
+    * property. Use via `import chisel3.util.ltl.Sequence.BoolSequence`.
+    */
+  implicit class BoolSequence(val inner: Bool) extends Sequence with SequenceAtom
+
+  /** Delay a sequence by a fixed number of cycles. Equivalent to `##delay` in
+    * SVA.
+    */
+  def delay(seq: Sequence, delay: Int = 1): Sequence = {
+    val ltl_delay = Module(new LTLDelayIntrinsic(delay, Some(0)))
+    ltl_delay.in := seq.inner
+    OpaqueSequence(ltl_delay.out)
+  }
+
+  /** Delay a sequence by a bounded range of cycles. Equivalent to `##[min:max]`
+    * in SVA.
+    */
+  def delayRange(seq: Sequence, min: Int, max: Int): Sequence = {
+    require(min <= max)
```

Comment:
I'm going to merge the PR anyway but in a future PR, any `require` really needs to have a message, otherwise it errors as just `Requirement Failed`.